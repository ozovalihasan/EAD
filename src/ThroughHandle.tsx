const ThroughHandle = ({visibility}: {visibility: string}) => (
  <svg className={`w-6 h-6 p-1 absolute -bottom-3 left-3/4 bg-slate-700 rounded-full -translate-x-1/2 ${visibility}`} viewBox="0 0 200 200">
    <path
      d="m 95,0 v 50 h 10 V 0"
      className="fill-slate-50"
    />
    <path
      d="m 95,75 v 50 H 105 v -50"
      className="fill-slate-50"
    />
    <path
      d="m 95,150 v 50 h 10 v -50"
      className="fill-slate-50"
    />
      
  </svg>
)

export default ThroughHandle;

