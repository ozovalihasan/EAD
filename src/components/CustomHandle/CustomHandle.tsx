import { Handle, Position } from 'reactflow';
import { useStore } from '@/zustandStore';
import { memo } from 'react';


export interface CustomHandleType {
  id: string, 
  nodeId: string, 
  handleType: string
}

export const CustomHandle = memo(({id, nodeId, handleType} : CustomHandleType) => {
  const onChangeAssociationType = useStore(store => store.onChangeAssociationType)
  
  return (
    
  <Handle
    onMouseDown={() => onChangeAssociationType(handleType, nodeId)}
    className="opacity-50 justify-center items-center border-none w-6 h-6 bottom-0 left-1/2"
    type="source"
    position={Position.Bottom}
    id={id}
  />
  )
})
