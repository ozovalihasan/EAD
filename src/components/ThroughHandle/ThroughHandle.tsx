import { CustomHandle } from '@/components';

export const ThroughHandle = ({nodeId} : {nodeId: string}) => {

  return (
    <div className="w-6 h-6 absolute left-3/4 bg-slate-700 rounded-full -translate-x-1/2 flex justify-center items-center" >
      <svg 
        className="stroke-[40] stroke-slate-50 w-3 h-3 "
        viewBox="0 0 200 200" 
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 100,20 V 40"
        />
        <path
          d="m 100,160 v 20"
        />
        <path
          d="m 100,90 v 20"
        />
      
      </svg>

      <CustomHandle 
        handleType='through' 
        nodeId={nodeId} 
        id="bottom3"
      />
      
    </div>
  )
}