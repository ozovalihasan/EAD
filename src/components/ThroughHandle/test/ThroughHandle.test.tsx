import { ThroughHandle } from '../ThroughHandle';
import { render, screen, act, renderHook, fireEvent } from "@testing-library/react";
import { EdgeTextProps } from 'react-flow-renderer';
import React from 'react';
import { CustomHandleType } from '../../CustomHandle/CustomHandle';



jest.mock('@/components',  () => ({
  CustomHandle: ({
    nodeId,
    id,
    handleType
  }: CustomHandleType) => {
    
    return (
      <>
        nodeId: {nodeId}
        id: {id}
        handleType: {handleType}
      </>
    );
  },
}));

let renderReadyComponent: JSX.Element;

beforeEach(() => {
  renderReadyComponent = (
    <ThroughHandle nodeId="1" />
  );
});

describe('<ThroughHandle />', () => {

  it('renders correctly', () => {
    const renderedContainer = render(renderReadyComponent );
    expect(renderedContainer).toMatchSnapshot();
  });

});