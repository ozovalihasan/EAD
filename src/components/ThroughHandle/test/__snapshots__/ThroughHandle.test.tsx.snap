// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThroughHandle /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-6 h-6 absolute left-3/4 bg-slate-700 rounded-full -translate-x-1/2 flex justify-center items-center"
      >
        <svg
          class="stroke-[40] stroke-slate-50 w-3 h-3 "
          viewBox="0 0 200 200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 100,20 V 40"
          />
          <path
            d="m 100,160 v 20"
          />
          <path
            d="m 100,90 v 20"
          />
        </svg>
        nodeId: 
        1
        id: 
        bottom3
        handleType: 
        through
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-6 h-6 absolute left-3/4 bg-slate-700 rounded-full -translate-x-1/2 flex justify-center items-center"
    >
      <svg
        class="stroke-[40] stroke-slate-50 w-3 h-3 "
        viewBox="0 0 200 200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 100,20 V 40"
        />
        <path
          d="m 100,160 v 20"
        />
        <path
          d="m 100,90 v 20"
        />
      </svg>
      nodeId: 
      1
      id: 
      bottom3
      handleType: 
      through
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
